src/session.ts

```diff
@@ -543,19 +642,25 @@ async function warmTransfer(
     this: WebPhoneSession,
     target: WebPhoneSession,
     transferOptions: any = {}
-): Promise<ReferClientContext> {
+): Promise<IncomingResponse> {
     await (this.localHold ? Promise.resolve(null) : this.hold());
-
-    await delay(300);
-
-    transferOptions.extraHeaders = (transferOptions.extraHeaders || []).concat(this.ua.defaultHeaders);
-
-    return this.refer(target, transferOptions);
+    return new Promise(resolve => {
+        transferOptions.extraHeaders = (transferOptions.extraHeaders || []).concat(this.ua.defaultHeaders);
+        transferOptions.receiveResponse = response => {
+            if (response.statusCode === 202) resolve(response);
+        };
+        this.logger.log('Completing warm transfer');
+        this.refer(target, transferOptions);
```

Comment:
Same here

Reply:
I was trying to resolve when I get response of sip 202 for refer. If this is not necessary I will just keep this as is and return ( Promise.resolve(  this.refer(target, transferOptions)) )